util/chunk/chunk.go

```diff
@@ -145,6 +148,54 @@ func (c *Chunk) AppendRow(colIdx int, row Row) {
 	}
 }
 
+// Copy copys rows ranged ["begin", "end") in "other" to current Chunk.
+func (c *Chunk) Copy(other *Chunk, begin, end int) {
+	for colID, src := range other.columns {
+		dst := c.columns[colID]
+		if src.isFixed() {
+			elemLen := len(src.elemBuf)
+			if len(dst.data) < elemLen*(end-begin) {
+				dst.data = make([]byte, elemLen*(end-begin), elemLen*(end-begin))
+			}
+			copy(dst.data, src.data[begin*elemLen:end*elemLen])
+			dst.data = dst.data[:elemLen*(end-begin)]
+		} else if src.isVarlen() {
+			beginOffset, endOffset := src.offsets[begin], src.offsets[end]
+			newDataLen := int(endOffset - beginOffset)
+			if len(dst.data) < newDataLen {
+				dst.data = make([]byte, newDataLen)
+			}
+			copy(dst.data, src.data[beginOffset:endOffset])
+			dst.data = dst.data[:newDataLen]
+
+			if len(dst.offsets) < end-begin+1 {
+				dst.offsets = make([]int32, end-begin+1)
+				dst.offsets[0] = 0
+			}
+			for i := begin; i < end; i++ {
+				dst.offsets[i-begin+1] = dst.offsets[i-begin] + src.offsets[i+1] - src.offsets[i]
+			}
+			dst.offsets = dst.offsets[:end-begin+1]
+		} else {
+			if len(dst.ifaces) < end-begin {
+				dst.ifaces = make([]interface{}, end-begin)
+			}
+			copy(dst.ifaces, src.ifaces[begin:end])
+			dst.ifaces = dst.ifaces[:end-begin]
+		}
+		dst.length, dst.nullCount = 0, 0
+		numBytesInBitmap := (end - begin + 7) >> 3
+		if len(dst.nullBitmap) < numBytesInBitmap {
+			dst.nullBitmap = make([]byte, numBytesInBitmap)
+		}
+		for i := begin; i < end; i++ {
```

Comment:
We should be able to append the bitmap bytes directly, in the case of not aligned, we can shift the bitmap.